src/layout/Table/TableComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { TableActionButton } from 'src/app-components/table/Table';
+import type { PropsFromGenericComponent } from 'src/layout';
+import type { IDataModelReference } from 'src/layout/common.generated';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+type TableComponentProps = PropsFromGenericComponent<'Table'>;
+
+type TableSummaryProps = {
+  componentNode: LayoutNode<'Table'>;
+};
+
+export function TableSummary({ componentNode }: TableSummaryProps) {
+  const item = useNodeItem(componentNode);
+  const { formData } = useDataModelBindings(item.dataModelBindings, 1, 'raw');
+  const { title } = item.textResourceBindings ?? {};
+  const { langAsString } = useLanguage();
+  const isMobile = useIsMobile();
+
+  const data = formData.simpleBinding;
+
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  if (data.length < 1) {
+    return null;
+  }
+  return (
+    <AppTable<IDataModelReference>
+      caption={title && <Caption title={<Lang id={title} />} />}
+      data={data}
+      columns={item.columnConfig.map((config) => ({
+        ...config,
+        header: langAsString(config.header),
+      }))}
+      mobile={isMobile}
+    />
+  );
+}
+
+export function TableComponent({ node }: TableComponentProps) {
+  const item = useNodeItem(node);
+  const { formData } = useDataModelBindings(item.dataModelBindings, 1, 'raw');
+  const removeFromList = FD.useRemoveFromListCallback();
+  const { title, description, help } = item.textResourceBindings ?? {};
+  const { langAsString } = useLanguage();
+  const { elementAsString } = useLanguage();
+  const accessibleTitle = elementAsString(title);
+  const isMobile = useIsMobile();
+
+  const data = formData.simpleBinding;
+
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  if (data.length < 1) {
+    return null;
+  }
+
+  const actionButtons: TableActionButton[] = [];
+
+  if (item.enableDelete) {
+    actionButtons.push({
+      onClick: (idx) => {
+        removeFromList({
+          startAtIndex: idx,
+          reference: {
+            dataType: item.dataModelBindings.simpleBinding.dataType,
+            field: item.dataModelBindings.simpleBinding.field,
+          },
+          callback: (_) => true,
+        });
+      },
+      buttonText: langAsString('general.delete'),
+      icon: <DeleteIcon />,
+      color: 'danger',
+    });
+  }
+
+  return (
+    <AppTable<IDataModelReference>
+      zebra={item.zebra}
+      size={item.size}
+      caption={
+        title && (
+          <Caption
+            title={<Lang id={title} />}
+            description={description && <Lang id={description} />}
+            helpText={help ? { text: <Lang id={help} />, accessibleTitle } : undefined}
+          />
+        )
+      }
+      data={data}
+      columns={item.columnConfig.map((config) => ({
+        ...config,
+        header: langAsString(config.header),
```

Comment:
Does it have to be a string, or could you pass `<Lang id={config.header} />` here instead?